src/System.Private.CoreLib/src/System/Threading/Tasks/DebuggerSupport.WinRT.cs

```diff
@@ -0,0 +1,100 @@
+        // This section of the class encapsulates the call to AsyncCausalityTracer for the async-aware callstacks.
+        //==============================================================================================================
+        public static bool LoggingOn
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                return _loggingOn;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceOperationCreation(CausalityTraceLevel traceLevel, Task task, String operationName, ulong relatedContext)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationCreation(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), operationName, relatedContext);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceOperationCompletion(CausalityTraceLevel traceLevel, Task task, AsyncStatus status)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationCompletion(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), status);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceOperationRelation(CausalityTraceLevel traceLevel, Task task, CausalityRelation relation)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationRelation(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), relation);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceSynchronousWorkStart(CausalityTraceLevel traceLevel, Task task, CausalitySynchronousWork work)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceSynchronousWorkStart(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), work);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceSynchronousWorkCompletion(CausalityTraceLevel traceLevel, CausalitySynchronousWork work)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceSynchronousWorkCompletion(traceLevel, CausalitySource.Library, work);
+            }
+        }
+
+        private static ulong OperationId(this Task task)
+        {
+            // The contract with the debugger is that the operation ID contains the appdomain ID in the upper 32 bits and Task.Id in the lower 32 bits.
+            // Project N does not have "appdomains" so we'll put a fixed "id" of 0x00000001 in the upper 32 bits.
+            return 0x0000000100000000LU | (ulong)(uint)(task.Id);
+        }
+
+        static DebuggerSupport()
```

Comment:
It would not make any difference. All (non-inlined) methods in the static class are accessing statics, and so they will have static ctor trigger anyway. I have fixed the lambda to use more compact syntax. Thanks